hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/ConsumerWorkloadDecoder.java

```diff
@@ -0,0 +1,43 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workload;
+
+import org.agrona.concurrent.UnsafeBuffer;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.consumers.subscription.id.SubscriptionId;
+import pl.allegro.tech.hermes.consumers.subscription.id.SubscriptionIds;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.sbe.stubs.AssignmentsDecoder;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.sbe.stubs.MessageHeaderDecoder;
+
+import java.util.HashSet;
+import java.util.Set;
+
+class ConsumerWorkloadDecoder {
+
+    private final SubscriptionIds subscriptionIds;
+
+    ConsumerWorkloadDecoder(SubscriptionIds subscriptionIds) {
+        this.subscriptionIds = subscriptionIds;
+    }
+
+    Set<SubscriptionName> decode(byte[] data) {
+        MessageHeaderDecoder header = new MessageHeaderDecoder();
+        AssignmentsDecoder body = new AssignmentsDecoder();
+
+        UnsafeBuffer buffer = new UnsafeBuffer(data);
+        header.wrap(buffer, 0);
+
+        if (header.templateId() != AssignmentsDecoder.TEMPLATE_ID) {
+            throw new IllegalStateException(String.format("AssignmentsDecoder TEMPLATE_ID=%d does not match encoded TEMPLATE_ID=%d",
```

Comment:
Do we handle/log this exception?

Reply:
Good question. It would be logged, but throwing this exception is not very safe, as it may break the workload calculation job if unable to decode data for just a single consumer, it's better to just log it and return empty response. The leader will override the corrupted data with new content.
I've changed the behaviour + added check for schema id, meaning schema id and message id should both match. 

Btw I've changed workload xml schema id to 2, as 1 was already used by max rate, and they should be unique. :) It's not very important as we read the messages from different sources/zk paths  but anyway it's now clear.